fake.go

```diff
@@ -23,20 +23,27 @@ import (
 	"regexp"
 	"sort"
 	"strings"
+	"sync"
 )
 
 // Fake is a fake implementation of Interface
 type Fake struct {
 	nftContext
+	// mutex is used to protect Table and LastTransaction.
+	// When Table and LastTransaction are accessed directly, the caller must acquire Fake.Rlock
+	// and release when finished.
+	sync.RWMutex
 
 	nextHandle int
 
 	// Table contains the Interface's table. This will be `nil` until you `tx.Add()`
 	// the table.
+	// Make sure to acquire Fake.Rlock before accessing Table in a concurrent environment.
 	Table *FakeTable
 
 	// LastTransaction is the last transaction passed to Run(). It will remain set until the
 	// next time Run() is called. (It is not affected by Check().)
+	// Make sure to acquire Fake.Rlock before accessing Table in a concurrent environment.
 	LastTransaction *Transaction
```

Comment:
```suggestion
	// Make sure to acquire Fake.Rlock before accessing LastTransaction in a concurrent environment.
	LastTransaction *Transaction
```

But that feels wrong anyway... `LastTransaction` is inherently non-concurrency-safe; you can only know that it has the transaction you want if you already know for sure what the most recent `Run()` call was. So maybe just add a warning about that. "This is probably not useful if Interface is being used concurrently." ?

And I guess that's an argument against `GetLastTransaction()`.

Reply:
I think it may be useful for concurrent usage also.
Let's say there is a controller thread running transactions, and test thread checking if it is correct.
On every change I make explicitly in the test, controller is expected to make exactly one transaction, but it may take time to reconcile. I will periodically check from the test thread `LastTransaction` until I see expected transaction, but to make it non-racy, I need to make sure I have RLock while reading LastTransaction.